common/reviews/api/tools.api.md

```diff
@@ -216,6 +220,17 @@ export abstract class AnnotationTool extends BaseTool {
     abstract toolSelectedCallback(evt: EventTypes_2.MouseDownEventType, annotation: Annotation, interactionType: InteractionTypes): void;
 }
 
+// @public (undocumented)
+type AnnotationVisibilityChangeEventDetail = {
+    added: Array<string>;
+    removed: Array<string>;
+    hidden: Array<string>;

```

Comment:
Why both hidden and visible"

Reply:
we agreed on using a var to represent the visible uids. But having both, for hidden and visible we can:
- Differentiate between being hidden or not existing
- We can do a show all, hide all
- The implementation is highly decoupled from the rest of code, which means we can easily tell how many are visible/hidden

Reply:
I still don't understand what the difference is - it seems like hidden and visible are the same type of thing.

Reply:
> We can do a show all, hide all.

Sure, but this is just toggling _one_ variable, as Bill says. I also don't see any benefit to having properties for these.

Reply:
@ladeirarodolfo I don't think I agree on having both hidden and visible, if that is what I communicated, I meant change everything to visible, visibility, isVisible

Reply:
ok to move forward lets consider the api for it. If is that what we want I can proceed, having just visible/isVisible.
Just to bring to our attention: an not visible annotation data (based on utility isAnnotationVisible) does not mean its hidden (it could be inexistent) 

export {
  getAnnotationUIDsVisible,
  getAnnotationUIDsVisibleCount,
  setAnnotationVisibility,
  isAnnotationVisible, 
  checkAndDefineIsVisibleProperty,
};

Reply:
> an not visible annotation data (based on utility isAnnotationVisible) does not mean its hidden (it could be inexistent) . I say that because on getState method, things might be very trick and inconsistent as far as annotation state combination is concerned

I need more context for this sentence, can you please elaborate? Why does a non visible annotation not mean it is hidden?

Reply:
Lets say by the context of this helper, it will only store visible uids.

If, for instance we invoke isAnnotationVisible(ABCUid) and this returns false.
It means:
1- We created ABCUid (it will be set as visible true) then we setVisibility false. Here isAnnotationVisible returning false is accurate.
2- There is no ABCUid annotation at all, isAnnotationVisible returning false is not consistent, because false might cause a wrong assumption the tool is hidden, but it actually does not exist at all (and we can not know that, if we only consider the proposed api).

Differently from locking, visible is actually the default state, but what we really want is to know if that is hidden or not, so with option 2 we might be lead to mistakes or false positives

Reply:
There doesn't need to be a "default" state, its just that we should change all the tools `addAnnotation` callback so that when an annotation is created the visible tag is created, and ad `visible: boolean` as a mandatory field on the `Annotation` type. Then we can just avoid all of the complex logic, it is `visible` (`true`) or not (`false`).

Reply:
An example for a possible wrong assumption would be the usage of getState method.

Lets say we change [this](packages/tools/src/stateManagement/annotation/config/getState.ts) to Visible instead.
Also for consistency lets say we only do the conditional `if(isVisible(ABCUID)` at end of clauses (prior to `AnnotationStyleStates.Default;`). So, the if there is no ABCUID, `isVisible(ABCUID)` is false, should we return here, should we go to other return (`AnnotationStyleStates.Default`)